tests/unit/test_instance_connection_manager.py

```diff
@@ -53,13 +55,99 @@ def test_InstanceConnectionManager_init(async_loop: asyncio.AbstractEventLoop) -
 
 
 @pytest.mark.asyncio
-async def test_InstanceConnectionManager_perform_refresh(
-    icm: InstanceConnectionManager, async_loop: asyncio.AbstractEventLoop
+async def test_perform_refresh_replaces_result(icm: InstanceConnectionManager) -> None:
+    """
+    Test to check whether _perform_refresh replaces a valid result with another valid result
+    """
+
+    class MockMetadata:
+        def __init__(self, expiration: datetime.datetime) -> None:
+            self.expiration = expiration
+
+    async def _get_metadata_success(*args: Any, **kwargs: Any) -> MockMetadata:
+        return MockMetadata(datetime.datetime.now() + datetime.timedelta(minutes=10))
```

Comment:
IMO, it's fine to move this outside of the individual tests since it's being used in multiple places